### PR TITLE
Move to Stripe Prices

### PR DESCRIPTION
**Pull request by dracos**
As well as being the modern Stripe way of doing it, this makes it easier to switch to different prices because we're not reliant on particular Stripe price (plan) IDs. A switch would then involve:
* Creating new Prices, including `calls` metadata entry
* Updating the price IDs in the settings and redeploying (for new subs)
* Emailing everyone to give them notice of the price change
* Running the script to schedule in everyone else's change after their next period (the end of the one after their current one)

Our Stripe test instance has new Prices, and the staging MapIt has example config in place with the right IDs, if you want to see how that's set up.

### CODE REVIEW

**Review by neprune — APPROVED** [review]

**Comment by dracos** [comment]
Was a bug in that if you modify a subscription with an items entry and not a subscription item ID, it adds to the subscription rather than replaces it (documented at https://docs.stripe.com/billing/subscriptions/upgrade-downgrade#changing but not in API docs I could see).

File: static/js/payment.js [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
+  price = price ? price.value : '';
   ctick = ctick ? ctick.checked : '';
   c = c ? c.value : '';
-  var num = 20;
-  if (plan === 'mapit-100k-v') {
-    num = 100;
-  } else if (plan === 'mapit-0k-v') {
-    num = 300;
-  }
+  var num = pricing.dataset[price] || 20;
```
Comment (dracos) [inline static/js/payment.js:11]: This was still hard-coded - had to add a minimum price data field, and data lookups are lowercase.

File: subscriptions/management/commands/schedule_price_change.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+stripe.api_key = settings.STRIPE_SECRET_KEY
+stripe.api_version = settings.STRIPE_API_VERSION
+
+
+class Command(BaseCommand):
+    help = "Update subscriptions on one Stripe Price to a different Price"
+
+    def add_arguments(self, parser):
+        prices = stripe.Price.list(limit=100)
+        price_ids = [price.id for price in prices.data if price.product.name.startswith('MapIt')]
```
Comment (dracos) [inline subscriptions/management/commands/schedule_price_change.py:16]: Had to expand data.product for this lookup to work

File: subscriptions/management/commands/schedule_price_change.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            phases[1]['discounts'] = [{'coupon': schedule.phases[0].discounts[0].coupon}]
+        stripe.SubscriptionSchedule.modify(schedule.id, phases=phases)
+
+    def handle(self, *args, **options):
+        old_price = options['old_price']
+        new_price = options['new_price']
+
+        for sub_obj in Subscription.objects.all():
+            subscription = stripe.Subscription.retrieve(sub_obj.stripe_id, expand=[
+                'schedule.phases.items.price'])
```
Comment (dracos) [inline subscriptions/management/commands/schedule_price_change.py:50]: Expanded discounts here, to ignore 100% off ones; plus inactive subs (there was one)

File: subscriptions/management/commands/schedule_price_change.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+            # Possibilities:
+            # * No schedule, just a monthly plan
+            # * 3 phases: This script has already been run
+            # * 2 phases: Just asked for downgrade, so at current price, then new price, then no schedule
+            # * 2 phases: Already on 'new price' schedule, awaiting change - nothing to do
+            # * 1 phase: Been downgraded, so new price, then no schedule
```
Comment (dracos) [inline subscriptions/management/commands/schedule_price_change.py:57]: This was my confusion thinking old parts of a phase disappeared, but they don't, so someone on a downgrade 2-phase will still have 2 phases when they're in the second phase. Reworked the below to handle this, also bug in price lookup check (needed to be price.id)

**Pull request merged**